arci/src/clients/joint_trajectory_clients_container.rs

```diff
@@ -207,4 +207,211 @@ mod tests {
+                Err(Error::Uninitialized {
+                    message: String::from("current pos is not exist."),
+                })
+            }
+            fn send_joint_positions(
+                &self,
+                _positions: Vec<f64>,
+                _duration: std::time::Duration,
+            ) -> Result<WaitFuture, Error> {
+                unimplemented!();
+            }
+            fn send_joint_trajectory(
+                &self,
+                _trajectory: Vec<TrajectoryPoint>,
+            ) -> Result<WaitFuture, Error> {
+                unimplemented!()
+            }
+        }
+        let clients = vec![
+            Dummy {
+                name: vec![String::from("part1"), String::from("high")],
+                pos: vec![1.0_f64, 3.0_f64],
+            },
+            Dummy {
+                name: vec![String::from("part2"), String::from("low")],
+                pos: vec![2.2_f64, 4.0_f64],
+            },
+        ];
+
+        let container = JointTrajectoryClientsContainer::new(clients);
+        let current = container.current_joint_positions();
+        let expect = vec![1.0_f64, 3.0_f64, 2.2_f64, 4.0_f64];
+
+        let positions = current.unwrap();
+
+        positions
+            .iter()
+            .zip(expect)
+            .for_each(|(pos, correct)| assert_approx_eq!(*pos, correct));
+    }
+
+    #[test]
+    fn test_container_send_pos_ok() {
+        use super::*;
+
+        #[derive(Debug, Clone)]
+        struct Dummy {
+            name: Vec<String>,
+            pos: Vec<f64>,
+        }
+        impl JointTrajectoryClient for Dummy {
+            fn joint_names(&self) -> &[String] {
+                &self.name
+            }
+            fn current_joint_positions(&self) -> Result<Vec<f64>, Error> {
+                Ok(self.pos.clone())
+            }
+            fn send_joint_positions(
+                &self,
+                _positions: Vec<f64>,
+                _duration: std::time::Duration,
+            ) -> Result<WaitFuture, Error> {
+                Ok(WaitFuture::ready())
+            }
+            fn send_joint_trajectory(
+                &self,
+                _trajectory: Vec<TrajectoryPoint>,
+            ) -> Result<WaitFuture, Error> {
+                unimplemented!()
+            }
+        }
+        let clients = vec![
+            Dummy {
+                name: vec![String::from("part1"), String::from("high")],
+                pos: vec![1.0_f64, 3.0_f64],
+            },
+            Dummy {
+                name: vec![String::from("part2"), String::from("low")],
+                pos: vec![2.2_f64, 4.0_f64],
+            },
+        ];
+
+        let container = JointTrajectoryClientsContainer::new(clients);
+        let pos = vec![3.4_f64, 5.8_f64, 0.1_f64, 2.5_f64];
+        let duration = std::time::Duration::from_secs(5);
+        let _ = container.send_joint_positions(pos, duration).unwrap();
```

Comment:
I want to use `await` to check wait logic is working.

```suggestion
        container.send_joint_positions(pos, duration).unwrap().await;
```

To do that, we need `#[tokio::test]` and `async` to the function name.
You can see the examples [here](https://github.com/openrr/openrr/blob/main/arci/tests/test_waits.rs)

Reply:
Is ` assert!(container.send_joint_positions(pos, duration).is_err());` bat test?

I want to know the difference between `unwrap test` and ` unwrap + await`.